substrate/frame/system/src/lib.rs

```diff
@@ -833,9 +841,14 @@ pub mod pallet {
 	#[pallet::event]
 	pub enum Event<T: Config> {
 		/// An extrinsic completed successfully.
-		ExtrinsicSuccess { dispatch_info: DispatchInfo },
+		ExtrinsicSuccess { weight: Weight, class: DispatchClass, pays: Pays },
 		/// An extrinsic failed.
-		ExtrinsicFailed { dispatch_error: DispatchError, dispatch_info: DispatchInfo },
+		ExtrinsicFailed {
+			dispatch_error: DispatchError,
+			weight: Weight,
+			class: DispatchClass,
+			pays: Pays,
```

Comment:
The type `DispatchInfo` is changing in this PR, `weight` field is removed, `call_weight` and `extension_weight` fields are introduced.

in `note_applied_extrinsic`, which emit this event, we don't have access to what was the `call_weight` and what was the `extension_weight`, we only have access to the `actual_weight` in post dispatch info.

If we want not to break, maybe we can introduce a specific type `EventDispatchInfo` or `AppliedInfo` with the exact same fields as the former `DispatchInfo`, i.e. `weight`, `class`, and `pays`?

Reply:
Yeah I actually realized this later as well and forgot about my comment here. I would go for a type that has the same fields as before. On the outside we are not interested in the extension weight any way, which should be static all the time.

Reply:
The extension weight is not static, it changes based on the transaction and the call possibly. `TransactionExtension::weight` takes a `&self` and a `call: &Call` to be able to provide an accurate weight at dispatch. This is because certain extensions that do authorization schemes would always need to charge the worst case weight, even when we have all of the information available to know the weight will actually be way smaller (for example, the `ChargeAssetTxPayment` extension charges less for native payments than for asset payments, and we know ahead of time whether we will try to swap or not).

Reply:
Yes, code is good to me, but if we use a new type with same fields as former `DispatchInfo` (i.e. weight, class, and pays), then all code outside the runtime making use of this event might be able to upgrade more smoothly.

Because, I guess, external code mostly make use of fields, not type names.

Reply:
I think if we change the most common events (`ExtrinsicSuccess` and `ExtrinsicFailed`) then indeed tons of integrations are going to break.  
I also think that we should keep the field names identical to make it easy for weakly-typed languages at lease.

Reply:
I introduced a type that is backwards compatible with the old struct and it will work with loosely typed languages.